dask/array/core.py

```diff
@@ -1597,6 +1596,8 @@ def elemwise(op, *args, **kwargs):
         except AttributeError:
             dt = None
 
+    name = kwargs.get('name') or 'elemwise-' + tokenize((op, args, dt))
```

Comment:
We'll need to ensure that `op` is rarely an interactively defined function.  The str representation of those will change each tme.